ddtrace/llmobs/_evaluators/runner.py

```diff
@@ -46,35 +52,40 @@ def start(self, *args, **kwargs):
     def on_shutdown(self):
         self.executor.shutdown()
 
-    def enqueue(self, span_event: Dict) -> None:
+    def enqueue(self, span_event: Dict, span: Span) -> None:
         with self._lock:
             if len(self._buffer) >= self._buffer_limit:
                 logger.warning(
                     "%r event buffer full (limit is %d), dropping event", self.__class__.__name__, self._buffer_limit
                 )
                 return
-            self._buffer.append(span_event)
+            self._buffer.append((span_event, span))
 
     def periodic(self) -> None:
         with self._lock:
             if not self._buffer:
                 return
-            events = self._buffer
+            span_batch = self._buffer  # type: List[Tuple[Dict, Span]]
             self._buffer = []
 
         try:
-            evaluation_metrics = self.run(events)
+            evaluation_metrics = self.run(span_batch)
             for metric in evaluation_metrics:
                 if metric is not None:
                     self._evaluation_metric_writer.enqueue(metric)
         except RuntimeError as e:
             logger.debug("failed to run evaluation: %s", e)
 
-    def run(self, spans):
+    def run(self, span_batch: List[Tuple[Dict, Span]]) -> List[Dict]:
         batches_of_results = []
 
         for evaluator in self.evaluators:
-            batches_of_results.append(self.executor.map(evaluator.evaluate, spans))
+            batches_of_results.append(
+                self.executor.map(
+                    evaluator.evaluate,
+                    [span_event for span_event, span in span_batch if self.sampler.sample(evaluator.label, span)],
```

Comment:
Does a sampler require an actual Span object?

Reply:
The `SamplingRule.sample` method takes in a `Span` object
```
def sample(self, span):
   ..
```
I thought the evaluator sampling rule could inherit from `SamplingRule` and just re-use that method
Do you see any downsides to storing a reference to the original `Span`?